flink-clients/src/main/java/org/apache/flink/client/web/JobSubmissionServlet.java

```diff
@@ -146,10 +146,22 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
 			}
 
 			String assemblerClass = null;
-			if (params.size() >= 2 && params.get(0).equals("assembler")) {
-				assemblerClass = params.get(1);
-				params.remove(0);
-				params.remove(0);
+			int pos = 0;
+			int parallelism = -1;
+			while(pos < params.size()) {
+				if (params.get(pos).equals("-c")) {
+					assemblerClass = params.get(pos + 1);
+					params.remove(pos);
+					params.remove(pos);
+				}
+				else if (params.get(pos).equals("-p")) {
+					parallelism = Integer.parseInt(params.get(pos + 1));
+					params.remove(pos);
+					params.remove(pos);
+				}
+				else {
+					pos++;

```

Comment:
If you change this to `break`, it will only accept these flags before the user program arguments (not mixed with them), which makes sense and reflects the way the command line handles it.